shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -235,7 +238,7 @@ - (void)setViewController:(FlutterViewController*)viewController {
   _viewController =
       viewController ? [viewController getWeakPtr] : fml::WeakPtr<FlutterViewController>();
   self.iosPlatformView->SetOwnerViewController(_viewController);
-  [self maybeSetupPlatformViewChannels];
+  [self maybeSetupChannelsHandler];
```

Comment:
Might be worth adding a unit test to FlutterEngine to make sure resourceChannel isn't nil after setting the view controller.